pkgs/development/tools/kustomize/4.nix

```diff
@@ -0,0 +1,48 @@
+{ lib, buildGoModule, fetchFromGitHub, installShellFiles }:
+
+buildGoModule rec {
+  pname = "kustomize_4";
+  version = "4.5.7";
+  # rev is the commit of the tag, mainly for kustomize version command output
+  rev = "56d82a8378dfc8dc3b3b1085e5a6e67b82966bd7";
+
+  ldflags = let t = "sigs.k8s.io/kustomize/api/provenance"; in
+    [
+      "-s"
+      "-X ${t}.version=${version}"
+      "-X ${t}.gitCommit=${rev}"
+    ];
+
+  src = fetchFromGitHub {
+    owner = "kubernetes-sigs";
+    repo = "kustomize";
+    rev = "kustomize/v${version}";
+    hash = "sha256-AHDUwXcYkI04nOBY8jScf+OE6k9Z5OqzhtWExK1rrKg=";
+  };
+
+  # avoid finding test and development commands
+  modRoot = "kustomize";
+  proxyVendor = true;
+  vendorHash = "sha256-9+k0Me5alZDNC27Mx0Q6vp0B2SEa+Qy0FoLSr/Rahkc=";
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    installShellCompletion --cmd kustomize \
+      --bash <($out/bin/kustomize completion bash) \
+      --fish <($out/bin/kustomize completion fish) \
+      --zsh <($out/bin/kustomize completion zsh)
+  '';
+
+  meta = with lib; {
+    description = "Customization of kubernetes YAML configurations";
+    longDescription = ''
+      kustomize lets you customize raw, template-free YAML files for
+      multiple purposes, leaving the original YAML untouched and usable
+      as is.
+    '';
+    homepage = "https://github.com/kubernetes-sigs/kustomize";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ carlosdagos vdemeester periklis zaninime Chili-Man saschagrunert ];
```

Comment:
why?

Reply:
I don't know. This is how the default `kustomize` and `kustomize_3` do things:

https://github.com/NixOS/nixpkgs/blob/c3297458d89dc39f5375dec39db975989ed702d0/pkgs/development/tools/kustomize/3.nix#L38

https://github.com/NixOS/nixpkgs/blob/c3297458d89dc39f5375dec39db975989ed702d0/pkgs/development/tools/kustomize/default.nix#L44

If there's a way I should change these, let me know.

Reply:
@AndersonTorres Please clarify if I should change anything here. This is my first contribution, so I need guidance.

Reply:
You are basically saying carlosdagos, vdemeester, periklis, zaninime, Chili-Man and saschagrunert are maintainers of this package.
Have you asked them to keep this?

Reply:
@carlosdagos, @vdemeester, @periklis, @zaninime, @Chili-Man, @saschagrunert: do any of you want to be removed from this new kustomize package?

@AndersonTorres is it appropriate to just start with me, and add folks back later? I don't want to _remove_ anyone who is expecting to listed here, but I'm gathering we view each of these as having their own set of maintainers.

Reply:
@abscondment   I'm fine either way and I suspect the others will probably reach the same way. I'm onboard with keeping line with what's already done to be consistent  so that there's less friction.

Reply:
If it's intentional to have diverging sets of maintainers for the versions then I'm happy to drop off the list.

Reply:
I am fine being a maintainer on both 👍🏼